### PR TITLE
public-key-package

### PR DESCRIPTION
**Pull request by jowparks**
The public key package with identifiers. 

Working version of the main ironfish sdk here: https://github.com/iron-fish/ironfish/pull/4757

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ ed25519-dalek = { version = "2.1.0", features = ["rand_core"] }
 once_cell = "1.8.0"
 rand_chacha = "0.3.1"
 rand_core = "0.6.4"
-reddsa = { git = "https://github.com/ZcashFoundation/reddsa.git", features = ["frost", "frost-rerandomized"] }
+reddsa = { git = "https://github.com/ZcashFoundation/reddsa.git", rev = "311baf8865f6e21527d1f20750d8f2cf5c9e531a", features = ["frost", "frost-rerandomized"] }
```
Comment (patnir) [inline Cargo.toml:19]: not sure why this change shows up. the current file is the same as main

File: src/multienc.rs [hunk 2]
```diff
@@ -65,7 +65,7 @@ where
         .map(|id| {
             let recipient_key = id.borrow().encryption_key();
             let shared_secret = agreement_secret.diffie_hellman(recipient_key).to_bytes();
-            let mut cipher = ChaCha20::new((&shared_secret).into(), (&nonce).into());
+            let mut cipher = ChaCha20::new((&shared_secret).into(), &nonce);
```
Comment (hughy) [inline src/multienc.rs:68]: are these just cleanup changes from unnecessary conversions?

File: Cargo.toml (hunk 1 above)
Comment (hughy) [reply Cargo.toml:19]: it is odd 🤷 

https://github.com/iron-fish/ironfish-frost/blob/2097825398c6a5ab0469058c52c2c2ed5842a3f0/Cargo.toml#L19

**Review by hughy — APPROVED** [review]

File: src/multienc.rs (hunk 2 above)
Comment (patnir) [reply src/multienc.rs:68]: Joe added these, I assume so
